dvc/data_cloud.py

```diff
@@ -6,9 +6,9 @@
 from dvc_data.db import get_index
 
 if TYPE_CHECKING:
+    from dvc_data.hashfile.db import HashFileDB
```

Comment:
Changed `ObjectDB` type to `HashFileDB`. I didn't change it elsewhere as it seems we'll have few more iterations on this, and it's better to use stricter version (i.e. `ObjectDB` where possible).